src/rotator_library/distributed_gate.py

```
@@ -0,0 +1,348 @@
+import socket
+import sqlite3
+import threading
+import time
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Dict, Optional, Tuple
+
+try:
+    import yaml  # already a project dep
+except Exception:  # pragma: no cover - yaml is present in prod
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)